fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NimbusMessagingMessageTest.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import android.content.Context
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.gleanplumb.CustomAttributeProvider
+import org.mozilla.fenix.helpers.HomeActivityIntentTestRule
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.nimbus.FxNimbus
+import org.mozilla.fenix.nimbus.Messaging
+
+/**
+ * This test is to test the integrity of messages hardcoded in the FML.
+ *
+ * It tests if the trigger expressions are valid, all the fields are complete
+ * and a simple check if they are localized (don't contain `_`).
+ */
+class NimbusMessagingMessageTest {
+    private lateinit var feature: Messaging
+    private lateinit var mDevice: UiDevice
```

Comment:
nit: device

Reply:
Leaving this as every other `androidTest` uses `mDevice`. Discussed with @t-p-white .